dotnet/src/SemanticKernel.Abstractions/Orchestration/KernelResultExtended.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.SemanticKernel.AI;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Generic Token counts record for exposing the token usage.
+/// </summary>
+/// <param name="CompletionTokens"></param>
+/// <param name="PromptTokens"></param>
+/// <param name="TotalTokens"></param>
+public sealed record TokenCounts(int CompletionTokens, int PromptTokens, int TotalTokens);
+
+/// <summary>
+/// Kernel result Strongly Typed Wrapper after execution.
+/// </summary>
+public record KernelResultExtended
+{
+    /// <summary>
+    /// Represents the result of the kernel execution.
+    /// </summary>
+    public string Result { get; init; }
+    /// <summary>
+    /// Represent the usage in Tokens.
+    /// </summary>
+    public TokenCounts? TokenCounts { get; set; }
```

Comment:
The reason why we didn't add things like `TokenCounts` or `ModelResults` to `FunctionResult`/`KernelResult` under `Microsoft.SemanticKernel.Orchestration` namespace is because these properties are related to AI execution. But there may be situations, when Kernel works only with native functions, so tokenization logic or model result is not applicable.

To make it more abstract, we added `Dictionary<string, object> Metadata` to `FunctionResult`, which can hold any kind of data, related to semantic or native function execution. We also provided `AIFunctionResultExtensions.GetModelResults(this FunctionResult result)` extension method under `Microsoft.SemanticKernel.AI` namespace, so it should help to get `ModelResults` in easier way. 

But in general, our goal was to provide minimal API, that is required to get complex results from Kernel. There are a lot of ways how to simplify it, by implementing wrappers, static classes, extension methods etc. But this is something that can be done on user side and users can choose what syntax works better in their specific scenarios. 

While we can keep the file with example to show how the result can be handled, I would try to keep `SemanticKernel.Abstractions` as lightweight as possible. 

Please let me know if that makes sense. Thank you!

Reply:
It does, thanks for the detailed explanation. And yes, it makes sense. Should I simplify this and put it as just an example on how to parse the KernelResult?

Reply:
@joslat Yes, you can put it as an example, I think it should be useful for developers. Thanks a lot!